discovery.go

```diff
@@ -1,78 +1,42 @@
 package discovery
 
 import (
-	"errors"
 	"fmt"
 	"net/url"
 	"os"
 	"strings"
-
-	kv "gopkg.in/clever/kayvee-go.v2"
 )
 
-// m is a convenience type for using kv.
-type m map[string]interface{}
-
 const (
-	templateVar = "SERVICE_%s_%s_%%s"
+	templateVar = "SERVICE_%s_%s_URL"
 )
 
-func getVar(envVar string) (string, error) {
-	envVar = strings.ToUpper(envVar)
-	envVar = strings.Replace(envVar, "-", "_", -1)
-	val := os.Getenv(envVar)
+// URLString finds the specified URL for a service based off of the service's name and which
+// interface you are accessing. Values are found in environment variables fitting the scheme:
+// SERVICE_{SERVICE NAME}_{INTERFACE NAME}_URL.
+func URLString(service, name string) (string, error) {
+	// build string
+	envKey := fmt.Sprintf(templateVar, service, name)
+
+	// standardize to uppercase & underscores
+	envKey = strings.Replace(strings.ToUpper(envKey), "-", "_", -1)
+
+	val := os.Getenv(envKey)
 	if val == "" {
-		return "", errors.New(kv.FormatLog("discovery-go", kv.Error, "missing env var", m{
-			"var": envVar,
-		}))
+		return "", fmt.Errorf("Missing env var, %s", envKey)
```

Comment:
I like the configure function, but would agree with others that we should not panic in this lib. Your example looks good to me. but I'd make it a little more structured. Why use strings when you can have better structure. 

Actually I'd just love to hear from you what the final state of ServiceConfig looks like?